lib/elixir/lib/module.ex

```diff
@@ -1514,6 +1514,66 @@ defmodule Module do
     end
   end
 
+  @doc """
+  Gets the last set value of a given attribute from a module.
+
+  If the attribute was marked with `accumulate` with
+  `Module.register_attribute/3`, the previous value to have been set will be
+  returned. If the attribute does not accumulate, this call is the same as
+  calling `Module.get_attribute/3`.
+
+  This function can only be used on modules that have not yet been compiled.
+  Use the `c:Module.__info__/1` callback to get all persisted attributes, or
+  `Code.fetch_docs/1` to retrieve all documentation related attributes in
+  compiled modules.
+
+  ## Examples
+
+      defmodule Foo do
+        Module.put_attribute(__MODULE__, :value, 1)
+        Module.get_last_attribute(__MODULE__, :value) #=> 1
+
+        Module.get_last_attribute(__MODULE__, :not_found, :default) #=> :default
+
+        Module.register_attribute(__MODULE__, :acc, accumulate: true)
+        Module.put_attribute(__MODULE__, :acc, 1)
+        Module.get_last_attribute(__MODULE__, :acc) #=> 1
+        Module.put_attribute(__MODULE__, :acc, 2)
+        Module.get_last_attribute(__MODULE__, :acc) #=> 2
+      end
+
+  """
+  @doc since: "1.15.0"
+  @spec get_last_attribute(module, atom, term) :: term
+  def get_last_attribute(module, key, default \\ nil) when is_atom(module) and is_atom(key) do
+    assert_not_compiled!(
+      {:get_last_attribute, 2},
+      module,
+      "Use the Module.__info__/1 callback or Code.fetch_docs/1 instead"
+    )
+
+    {set, bag} = data_tables_for(module)
+
+    case :ets.lookup(set, key) do
+      [{_, _, :accumulate, _}] ->
+        case :ets.select(bag, [{{{:accumulate, key}, :"$1"}, [], [:"$1"]}], 1) do
+          {[value], _} -> value
+          _ -> default
+        end
+
+      # This check ensures the same behaviour as `get_attribute/3`,
+      # which returns the default when the attribute is set to `nil`.
```

Comment:
You are right. Let's go ahead and fix this in this PR as well? Thank you! :heart: